sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -6573,6 +6575,100 @@ class AstBuilder extends DataTypeAstBuilder
     )
   }
 
+  /**
+   * Create a [[DeclareCursor]] command wrapped in SingleStatement.
+   *
+   * For example:
+   * {{{
+   *   DECLARE cursor_name CURSOR FOR SELECT * FROM table;
+   * }}}
+   */
+  override def visitDeclareCursorStatement(
+      ctx: DeclareCursorStatementContext): LogicalPlan = withOrigin(ctx) {
+    val cursorName = ctx.strictIdentifier().getText
+    // Extract original SQL text to preserve parameter markers
+    val queryText = getOriginalText(ctx.query())
+
+    // Use a placeholder plan (LocalRelation) - the actual query will be parsed at execution time
+    // This prevents parameter markers from being resolved during script analysis
+    val query: LogicalPlan = LocalRelation()
```

Comment:
why do we need a LogicalPlan field in `DeclareCursor`?

Reply:
No reason.... fixed, switched to LeafCommand as well.